src/main/java/uk/nhs/ctp/tkwvalidation/model/FhirMessageAudit.java

```diff
@@ -0,0 +1,88 @@
+package uk.nhs.ctp.tkwvalidation.model;
+
+import static org.apache.commons.lang3.StringUtils.defaultIfEmpty;
+import static org.springframework.http.MediaType.parseMediaType;
+
+import java.net.URI;
+import java.time.Instant;
+import java.util.Arrays;
+import java.util.List;
+import lombok.Builder;
+import lombok.Value;
+import org.apache.http.HttpHeaders;
+import org.springframework.http.MediaType;
+import uk.nhs.ctp.audit.AuditParser;
+import uk.nhs.ctp.audit.model.AuditEntry;
+import uk.nhs.ctp.audit.model.AuditSession;
+
+@Value
+@Builder
+public class FhirMessageAudit {
+  private static final List<MediaType> FHIR_FORMATS = Arrays.asList(
+      parseMediaType("application/fhir+json"),
+      parseMediaType("application/fhir+xml"),
+      parseMediaType("application/json+fhir"),
+      parseMediaType("application/xml+fhir")
+  );
+
+  String filePath;
+  String requestBody;
+  String responseBody;
+  Instant moment;
+
+  public static FhirMessageAudit from(AuditSession session, String basePath, boolean includeRequestBody) {
+    var requestBody = includeRequestBody && isFhirContent(session.getRequestHeaders())
```

Comment:
Just so I understand, what's the behavior here if it's _not_ fhir content? Do we still end up sending a FhirMessageAudit just with no body?

Reply:
If neither the request nor the response hold fhir content the `AuditSelector` ends up passing along a `FhirMessageAudit` with `null` `requestBody` and `responseBody`, yes, and then the `AuditZipBuilder` takes care to not include null body entries in the .zip file.
Agreed regarding the testing, will update soon.

Reply:
Just seems strange is all, why send anything at all, since the requirement is to only send FHIR messages?

Reply:
The requirement was all audits until that caused problems with the TKW parsing non-fhir messages. It's debatable whether we should be _auditing_ non-fhir messages, but we have agreed no send them to TKW